app/components/download-spreadsheet.js

```diff
@@ -20,173 +28,105 @@ export default Component.extend({
       const profile = this.get('data');
       const mode = this.get('mode');
       const tab = this.get('tab');
+      const excludedProperties = this.get('excludedProperties');
+      const censusCurrent = this.get('censusCurrent');
+      const censusChange = this.get('censusChange');
+      const acsCurrent = this.get('acsCurrent');
+      const acsChange = this.get('acsChange');
 
+      // profileForPrint is set to an empty array at beginning of each profile.map
+      const profileForPrint = [];
+      // current view AND tab is NOT census
       if (mode === 'current') {
         if (tab !== 'profile.census') {
           profile.map((row) => {
-            const truncatedRow = row;
-            delete truncatedRow.codingThresholds;
-            delete truncatedRow.rowConfig;
-            delete truncatedRow.notinprofile;
-            delete truncatedRow.variablename;
-            delete truncatedRow.year;
-            delete truncatedRow.is_most_recent;
-            delete truncatedRow.geotype;
-            delete truncatedRow.producttype;
-            delete truncatedRow.release_year;
-            delete truncatedRow.unittype;
-            delete truncatedRow.special;
-            delete truncatedRow.previous_sum;
-            delete truncatedRow.previous_m;
-            delete truncatedRow.previous_cv;
-            delete truncatedRow.previous_percent;
-            delete truncatedRow.previous_percent_m;
-            delete truncatedRow.previous_is_reliable;
-            delete truncatedRow.change_sum;
-            delete truncatedRow.change_m;
-            delete truncatedRow.change_significant;
-            delete truncatedRow.change_percent;
-            delete truncatedRow.change_percent_m;
-            delete truncatedRow.change_percent_reliable;
-            delete truncatedRow.change_percentage_point;
-            delete truncatedRow.change_percentage_point_m;
-            delete truncatedRow.change_percentage_point_reliable;
-            return truncatedRow;
+            // remove properties that are listed in excludedProperties array
+            excludedProperties.forEach((item) => {
+              delete row[item];
+            });
```

Comment:
Non-blocking nit: 

I remember you mentioned that deleting things causes issues for _subsequent_ views. Is this still true?

In JS, you pass information by _reference_, which means that we're actually deleting something in memory here, which may cascade into other parts of the app. I'm not sure if that's still true for this, but it's worth checking. 

An alternative would be to use `getProperties`, maintaining lists of things to include.